### PR TITLE
Fix requirements traceability issues in azure-c-util

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Fix requirements traceability issues in azure-c-util

### CODE REVIEW

File: adapters/lock_win32.c [hunk 1]
```diff
@@ -31,7 +31,7 @@ LOCK_RESULT Lock_Deinit(LOCK_HANDLE handle)
     LOCK_RESULT result;
     if (handle == NULL)
     {
-        /* Codes_SRS_LOCK_10_007: [Lock_Deinit on NULL handle passed returns LOCK_ERROR] */
+        /* Codes_SRS_LOCK_10_013: [ `Lock_Deinit` on NULL `handle` passed returns `LOCK_ERROR` ]*/
```
Comment (mattdurak) [inline adapters/lock_win32.c]: >[ ` [](start = 34, length = 3)

nit: backticks

File: devdoc/crt_abstraction_requirements.md [hunk 2]
```diff
@@ -94,6 +92,8 @@ int sprintf_s(char* dst, size_t dstSizeInBytes, const char* format, ... );
 
 **SRS_CRT_ABSTRACTIONS_99_027: [** sprintf_s shall return the number of characters stored in dst upon success.  This number shall not include the terminating null character. **]**
 
+**SRS_CRT_ABSTRACTIONS_02_006: [** If the conversion fails for any reason (for example, insufficient buffer space), a non-zero return value shall be supplied and size_tToString shall fail. **]**
```
Comment (mattdurak) [inline devdoc/crt_abstraction_requirements.md]: >size_tToString [](start = 162, length = 14)

`sprintf_s` ?

File: tests/lock_ut/lock_ut.c [hunk 3]
```diff
@@ -72,6 +72,7 @@ TEST_FUNCTION_CLEANUP(cleans)
 }
 
 /* Tests_SRS_LOCK_10_002: [Lock_Init on success shall return a valid lock handle which should be a non NULL value] */
+/* Tests_SRS_LOCK_10_004: [ `Lock` shall be implemented as a non-recursive lock ]*/
```
Comment (mattdurak) [inline tests/lock_ut/lock_ut.c]: >` [](start = 28, length = 1)

nit: backticks

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -5,6 +5,7 @@ if(${run_unittests})
     add_subdirectory(agenttime_ut)
     add_subdirectory(azure_base64_ut)
     add_subdirectory(buffer_ut)
+    add_subdirectory(constbuffer_ut)
```
Comment (mattdurak) [inline tests/CMakeLists.txt]: >constbuffer_ut [](start = 21, length = 14)

interesting, wasn't even building

File: adapters/lock_win32.c (hunk 1 above)
Comment (dcristoloveanu) [reply adapters/lock_win32.c]: Fixed.

---
In reply to: [362558185](https://github.com/Azure/azure-c-util/pull/7#discussion_r362558185) [](ancestors = 362558185)

File: devdoc/crt_abstraction_requirements.md (hunk 2 above)
Comment (dcristoloveanu) [reply devdoc/crt_abstraction_requirements.md]: Fixed.

---
In reply to: [362558963](https://github.com/Azure/azure-c-util/pull/7#discussion_r362558963) [](ancestors = 362558963)

File: tests/CMakeLists.txt (hunk 4 above)
Comment (dcristoloveanu) [reply tests/CMakeLists.txt]: Yep, missed it when moving things the first time.

**Pull request merged**